components/studio/bin/hab-studio.sh

```diff
@@ -909,6 +909,20 @@ exit_with() {
   exit $2
 }
 
+# **Internal** Builds up a secret environmt based on the prefix `HAB_STUDIO_SECRET_`
+# to pass into the studio
+sanitize_secrets() {
+  blacklist="HAB_STUDIO_SECRET_HAB_BINLINK_DIR HAB_STUDIO_SECRET_HAB_ORIGIN HAB_STUDIO_SECRET_HOME HAB_STUDIO_SECRET_LC_ALL HAB_STUDIO_SECRET_PATH HAB_STUDIO_SECRET_PWD HAB_STUDIO_SECRET_STUDIO_TYPE HAB_STUDIO_SECRET_TERM HAB_STUDIO_SECRET_TERMINFO"
```

Comment:
Please consider using `blocklist` or something else as `blacklist`/`whitelist` can be offensive.

Additionally, you can eliminate the need for a variable altogether since `unset` can take many arguments:
```bash
  unset HAB_STUDIO_SECRET_HAB_BINLINK_DIR HAB_STUDIO_SECRET_HAB_ORIGIN \
        HAB_STUDIO_SECRET_HOME HAB_STUDIO_SECRET_LC_ALL HAB_STUDIO_SECRET_PATH \
        HAB_STUDIO_SECRET_PWD HAB_STUDIO_SECRET_STUDIO_TYPE \
        HAB_STUDIO_SECRET_TERM HAB_STUDIO_SECRET_TERMINFO
```

Reply:
Another alternative would be to omit the common prefix to make it more readable:
```bash
for x in HAB_BINLINK_DIR HAB_ORIGIN HOME LC_ALL PATH PWD STUDIO_TYPE TERM TERMINFO; do
  unset "HAB_STUDIO_SECRET_$x"
done
```